src/app_size_analyzer/cli.py

```diff
@@ -0,0 +1,273 @@
+            console=console,
+            disable=quiet,
+        ) as progress:
+            task = progress.add_task("Analyzing iOS app bundle...", total=None)
+
+            analyzer = IOSAnalyzer(
+                working_dir=working_dir,
+                skip_swift_metadata=skip_swift_metadata,
+                skip_symbols=skip_symbols,
+            )
+            results = analyzer.analyze(input_path)
+
+            progress.update(task, description="Analysis complete!")
+
+        end_time = time.time()
+        duration = end_time - start_time
+
+        results = results.model_copy(update={"analysis_duration": duration})
+
+        if output_format == "json":
+            _write_json_output(results, output, quiet)
+        else:
+            _print_table_output(results, quiet)
+
+        if not quiet:
+            console.print(f"\n[bold green]✓[/bold green] Analysis completed in {duration:.2f}s")
+            _print_summary(results)
+
+    except Exception as e:
+        if verbose:
+            console.print_exception()
+        else:
+            console.print(f"[bold red]Error:[/bold red] {e}")
+        raise click.Abort()
+
+
+@cli.command()
+@click.argument("input_path", type=click.Path(exists=True, path_type=Path), metavar="INPUT_PATH")
+@click.option(
+    "-o",
+    "--output",
+    type=click.Path(path_type=Path),
+    default="android-analysis-report.json",
+    help="Output path for the JSON analysis report.",
+    show_default=True,
+)
+def android(input_path: Path, output: Path) -> None:
+    """Analyze an Android app bundle and generate a size report.
+
+    INPUT_PATH can be:
+    - Android .apk file
+    - Android .aab file
+
+    [Coming Soon - Android analysis is not yet implemented]
+    """
+    console.print("[bold red]Android analysis is not yet implemented.[/bold red]")
+    console.print("This feature is coming soon!")
+    raise click.Abort()
+
+
+def _validate_ios_input(input_path: Path) -> None:
+    """Validate that the input path looks like an iOS artifact."""
+    suffix = input_path.suffix.lower()
+    valid_extensions = {".zip"}
+
+    if suffix not in valid_extensions:
+        raise click.BadParameter(
+            f"'{input_path}' doesn't look like a typical iOS artifact. "
+            f"Expected one of: {', '.join(sorted(valid_extensions))}"
+        )
+
+
+def _write_json_output(results: AnalysisResults, output_path: Path, quiet: bool) -> None:
+    """Write results to JSON file."""
+    # Ensure output directory exists
+    output_path.parent.mkdir(parents=True, exist_ok=True)
+
+    # Write JSON with proper formatting
+    with open(output_path, "w", encoding="utf-8") as f:
+        json.dump(results.to_dict(), f, indent=2, ensure_ascii=False)
+
+    if not quiet:
+        console.print(f"[bold green]✓[/bold green] Results written to: [cyan]{output_path}[/cyan]")
+
+
+def _print_table_output(results: AnalysisResults, quiet: bool) -> None:
```

Comment:
I'll probably delete this table output but so far it's been helpful with debugging.